news/_posts/2016-05-10-prestashop-coding-standards.md

```diff
@@ -0,0 +1,191 @@
+
+* Use camelCase, not_underscores, for variable names, function names, method names, and arguments names.
+* <img src="/assets/images/2016/05/yoda.gif" style="display:inline;width:24px;vertical-align:top;"> [Yoda conditions](https://en.wikipedia.org/wiki/Yoda_conditions) you should use.
+* Add a comma after each array item in a multi-line array, even after the last one.
+
+{% highlight php startinline=true %}
+$defaultOptions = array(
+    'some_default' => 'values',
+    'another_default' => 'more values',
+);
+
+if (true === isset($defaultOptions['some_default'])) {
+    return;
+}
+{% endhighlight %}
+
+More details and a complete example can be found on [Symfony documentation](http://symfony.com/doc/current/contributing/code/standards.html).
+
+### JavaScript
+
+For our JavaScript code, we chose to follow the well-detailed and documented [Airbnb JavaScript Style Guide](https://github.com/airbnb/javascript). It respects the ECMAScript 6 style, and Babel will transpile it into understandable scripts for the browsers.
+
+Some significant points:
+
+* Always use semicolons at the end of an instruction.
+* Use soft tabulations with two spaces to indent.
+* Use braces with all multi-line blocks.
+
+{% highlight javascript startinline=true %}
+(function () {
+  if (test) {
+    return false;
+  }
+}());
+{% endhighlight %}
+
+Do not hesitate to look at their [complete documentation](https://github.com/airbnb/javascript).
+
+### CSS & HTML
+
+Regarding the CSS and HTML codebase, because we have integrated Bootstrap 4 both in the back office and front office, we have decided to follow their [coding standards](http://codeguide.co), maintained by @mdo.
+
+When you are writing HTML code, if your line is longer than 120 characters, we have chosen to do a line break after each attributes with an indentation. For example, it will look like this:
+
+{% highlight html startinline=true %}
+<img class="test"
+  id="..."
+  name="..."
+  data="..."
+  src="..."
+  title="..."
+  alt="..."
+  role="..."
+>
+{% endhighlight %}
+
+Some significant points:
+
+* HTML
+  * Use soft tabulations with two spaces to indent.
+  * Avoid superfluous HTML markup.
+  * No more self-closing tag.
+* CSS
+  * Use soft tabulations with two spaces to indent.
+  * When grouping selectors, keep individual selectors to a single line.
+  * End all declarations with a semicolon, even the last one.
+
+{% highlight html startinline=true %}
+<img class="test" src="...">
+{% endhighlight %}
+
+{% highlight css startinline=true %}
+img,
+a,
+input[type="text"] {
+  display: block;
+  padding: 15px;
+  margin-bottom: 15px;
+  background-color: rgba(0,0,0,.5);
+  box-shadow: 0 1px 2px #ccc, inset 0 1px 0 #fff;
+}
+{% endhighlight %}
+
+### Smarty / Twig
+
+PrestaShop 1.7 uses both the Smarty and Twig template engine. Because they are mostly used to generate HTML, we ask that Smarty/Twig code be written following the HTML standards above -- as well as the one used by SensioLabs for Twig.
```

Comment:
From Laurène:
PrestaShop 1.7 uses both the Smarty and Twig template engine. Since those are mostly used to generate HTML, we expect the Smarty/Twig code to comply with the HTML standards above -- as well as the one used by SensioLabs for Twig.